nservicebus/sqlserver/configuration.md

```diff
@@ -0,0 +1,178 @@
+
+snippet:sqlserver-config-native-transactions
+
+When in this mode, the receive operation is wrapped in a plain ADO.NET `SqlTransaction`. Both connection and the transaction instances are attached to the pipeline context under these keys `SqlConnection-{ConnectionString}` and `SqlTransaction-{ConnectionString}` and are available for user code so that the updates to user data can be done atomically with queue receive operation.
+
+
+### No transaction
+
+The no transaction mode requires `Transactions.Enabled` to be set to false which can be achieved via following API call:
+
+snippet:sqlserver-config-no-transactions
+
+When in this mode, the receive operation is not wrapped in any transaction so it is executed by the SQL Server in its own implicit transaction.
+
+WARNING: This means that as soon as the `DELETE` operation used for receiving completes, the message is gone and any exception that happens during processing of this message causes it to be permanently lost.
+
+### [Entity Framework](https://msdn.microsoft.com/en-us/data/ef.aspx) caveats
+
+Sharing the same connection string is easy when you have full control over the creation of the underlying `SQL connection`, but it can be problematic when dealing with entities based on the [Entity Framework ADO.Net Data Model (EDMX)](https://msdn.microsoft.com/en-us/library/vstudio/cc716685.aspx).
+
+The `DbContext` generated by Entity Framework does not expose a way to inject a simple database connection string; the underlying problem is that Entity Framework requires an `Entity Connection String` that has much more information than a simple connection string.
+
+It is possible to generate a custom a custom `EntityConnection` and inject it into the Entity Framework `DbContext` instance:
+
+snippet:EntityConnectionCreationAndUsage
+
+In the above snippet you are using `EntityConnectionStringBuilder` class to create a valid `Entity Connection String`, once you have that you can create a new `EntityConnection` instance.
+The `DbContext` generated by default by Entity Framework does not have a constructor that accepts an `EntityConnection` as a parameter but since it is a partial class you can easily add one:
+
+snippet:DbContextPartialWithEntityConnection
+
+NOTE: The above snippets assumes that the created entity data model is named `MySample`, change all the reference to match your project conventions.
+
+
+## Callbacks
+
+### Disable callbacks
+
+Callbacks and consequently, callback queues (secondary queues) receivers are enabled by default. In order to disable them, one must use the configuration API:
+
+snippet:sqlserver-config-disable-secondaries
+
+Secondary queues use same adaptive concurrency model to the primary queue. Secondary queues (and hence callbacks) are disabled for satellite receivers.
+
+
+### Callback Receiver Max Concurrency
+
+Changes the number of threads that should be used for the callback receiver. The default is 1 thread.
```

Comment:
Question - is it just me or it's not really clear what are "satellite receivers" and "satellite queues"? Is it something we should assume reader just knows?